imageroot/actions/add-user/50add_user

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (C) 2021 Nethesis S.r.l.
+# http://www.nethesis.it - nethserver@nethesis.it
+#
+# This script is part of NethServer.
+#
+# NethServer is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License,
+# or any later version.
+#
+# NethServer is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see COPYING.
+#
+
+import agent
+import json
+import sys
+import subprocess
+import os
+
+request = json.load(sys.stdin)
+user = request['user']
+groups = request['groups']
+password = request.get('password', '') # Empty string implies a random password is set
+full_name = request['full_name']
+locked = request.get('locked', False)
+
+sambatool_cmd = ['podman', 'exec', '-i', os.environ['MODULE_ID'], 'samba-tool']
+adduser_cmd = sambatool_cmd + ['user', 'create', user]
+
+if not password:
+    adduser_cmd += ['--random-password']
```

Comment:
Is there a way to recover the random password?
I guess the admin just need to set it at later time.

Reply:
Yes the admin must set it lately. I'm thinking about bulk users creation: Samba wants a complex password is set and provides `--random-password` to initialize the new user entry properly. UI is expected to pass a non-empty string to satisfy the JSON schema validator. The attribute is optional for bulk operations.